src/appshell/internal/startupscenario.cpp

```diff
@@ -64,39 +65,39 @@ void StartupScenario::setStartupScorePath(const io::path& path)
 
 void StartupScenario::run()
 {
+    TRACEFUNC;
+
     if (!m_startupScorePath.empty()) {
         openScore(m_startupScorePath);
         return;
     }
 
-    async::Async::call(nullptr, [this]() {
-        StartupSessionType sessionType;
-        if (!m_sessionType.isEmpty()) {
-            sessionType = sessionTypeTromString(m_sessionType);
-        } else {
-            sessionType = configuration()->startupSessionType();
-        }
-
-        interactive()->open(startupPageUri(sessionType));
-
-        switch (sessionType) {
-        case StartupSessionType::StartEmpty:
-            break;
-        case StartupSessionType::StartWithNewScore:
-            dispatcher()->dispatch("file-new");
-            break;
-        case StartupSessionType::ContinueLastSession:
-            dispatcher()->dispatch("continue-last-session");
-            break;
-        case StartupSessionType::StartWithScore: {
-            openScore(configuration()->startupScorePath());
-        } break;
-        }
-
-        if (!configuration()->hasCompletedFirstLaunchSetup()) {
-            runFirstLaunchSetup();
-        }
-    });
+    StartupSessionType sessionType;
+    if (!m_sessionType.isEmpty()) {
+        sessionType = sessionTypeTromString(m_sessionType);
+    } else {
+        sessionType = configuration()->startupSessionType();
+    }
+
+    interactive()->open(startupPageUri(sessionType));
+
+    switch (sessionType) {
```

Comment:
It looks like these actions in the switch should be performed after the page is opened, that is, an excellent solution would be to add a signal (channel) to interactive, which will notify that the page has opened and make these actions to the handlers of this notification. Well, or at least make them asynchronously at the next event loop.

Reply:
OK, I'll add a signal, but I want to do this in a separate PR